Create changelog workflow

tyrion70 opened the pull request:
This workflow checks that the CHANGELOG file was updated (or created) in each PR.

kaleofduty commented:
Looks like the run errored?

kaleofduty commented on .github/workflows/changelog.yml:
```diff
@@ -0,0 +1,31 @@
+    steps:
+      - name: Check for changed files
+        id: changedfiles
+        uses: futuratrepadeira/changed-files@88878571ea9b265011eb41efe51c322e75e803c0
```
Can you add comments with version numbers?

kaleofduty commented on .github/workflows/changelog.yml:
```diff
@@ -0,0 +1,31 @@
+          repo-token: ${{ github.token }}
+          pattern: '^CHANGELOG$'
+      - name: Make a comment
+        uses: unsplash/comment-on-pr@182ec73bcf2b77868dcadd16987721fcfaca76d9
```
Can you add comments with version numbers?

tyrion70 commented:
> Looks like the run errored?

Yeah I updated the other actions to a newer version but that broke so I reverted back to the old version I was using. Also added the version numbers now in the comments

kaleofduty commented on .github/workflows/changelog.yml:
```diff
@@ -0,0 +1,31 @@
+  pull_request:
+    branches: [ develop ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
```
I think we can get rid of this comment

kaleofduty commented on .github/workflows/changelog.yml:
```diff
@@ -0,0 +1,31 @@
+# events but only for the master branch
+on:
+  pull_request:
+    branches: [ develop ]
```
Might make sense to run it on all PRs, not just into develop?

j16r replied:
Might want to make sure that doesn't block releases/hotfixes etc. which don't necessarily add any new changes.

j16r commented:
If I'm doing something that shouldn't go into the CHANGELOG, is there a convenient escape hatch? e.g: a chore / refactor.

tyrion70 commented:
> If I'm doing something that shouldn't go into the CHANGELOG, is there a convenient escape hatch? e.g: a chore / refactor.

The PR is not blocked from merging, its just a suggestion to add information. You could just ignore that but it serves as a reminder for reviewers and the PR starter to at least consider it. Is that OK?

kaleofduty commented:
> If I'm doing something that shouldn't go into the CHANGELOG, is there a convenient escape hatch? e.g: a chore / refactor.

You ignore the comment posted by the bot? It's just meant as a reminder to use your judgement whether it would make sense to add a changelog item.

j16r commented:
@kaleofduty @tyrion70 yeah it wasn't clear if this was going to be blocking or suggestive, happy if it's just a comment on the PR.

kaleofduty commented:
Perhaps we should edit the message to make that more clear. E.g.
"I see that you haven't updated any CHANGELOG files. Would it make sense to do so?"

tyrion70 commented:
That makes sense, I'll change the message to be more clear there.. @j16r do you like the format for the CHANGELOG-explorer there btw? I haven't added any historical data except the last release. Open for suggestions there.

j16r commented:
Yeah it looks good, also thank you for using iso8601.
